src/Infrastructure.Dapper/Repositories/UserRepository.cs

```diff
@@ -48,6 +48,27 @@ public async Task<User> GetByEmailAsync(string email)
         }
     }
 
+    public async Task<IEnumerable<User>> GetManyByEmailsAsync(IEnumerable<string> emails)
+    {
+        var emailTable = new DataTable();
+        emailTable.Columns.Add("Email", typeof(string));
+        foreach (var email in emails)
+        {
+            emailTable.Rows.Add(email);
+        }
+
+        using (var connection = new SqlConnection(ConnectionString))
+        {
+            var results = await connection.QueryAsync<User>(
+                $"[{Schema}].[{Table}_ReadByEmails]",
+                new { Emails = emailTable.AsTableValuedParameter("dbo.EmailArray") },
```

Comment:
🌱 I get that this type was already there, but I am wondering about its value as such a structured concept and a type vs. a simple generic string array that has reuse (and to ignore field lengths).

Reply:
Fair. I'll probably leave that as a future enhancement if that's alright? 

However, I will ask: what types of issues can arise from allowing any length strings vs scoping the strings to a specific domain which matche the length of that domain in our DB (ex: email is NVARCHAR 256)?  If a string longer than a DB column is passed in will it throw an error or truncate the data?

Reply:
When you try to update an NVARCHAR(256) field with a value from a variable that is longer the that the actual value in the variable is cut off. SQL Server will truncate the value to fit into the NVARCHAR(300) field. This means that only the first 300 characters of the string will be saved in the NVARCHAR(300) field, and the rest will be lost.

Reply:
Ah, good to know! So, it's more safe to use the explicit length types for these operations.

Reply:
Can certainly be future work. If there's nothing above the database call that's checking this length then this is gonna happen anyway, so it seems more reusable to get rid of this constricted passthrough type and use something open since the expectation is sanitization at a higher level.

Reply:
Agreed on an open approach. In this case I had ANSI_warnings off for my session due to some previous testing , it is on by default 

I set it to ON and tried inserting to a tvp and a table with max nvarchar(256)

Both gave an error and aborted:

set ANSI_WARNINGS ON

DECLARE @EmailList AS dbo.EmailArray;
   

INSERT INTO email (Email) VALUES ('email1aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaadddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddaaaaaaaaaaaaaaa@example.com');


INSERT INTO @EmailList (Email) VALUES ('email1aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaadddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddaaaaaaaaaaaaaaa@example.com');


Msg 2628, Level 16, State 1, Line 6
String or binary data would be truncated in table 'vault_dev.dbo.email', column 'email'. Truncated value: 'email1aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaadddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd'.
The statement has been terminated.
Msg 2628, Level 16, State 1, Line 9
String or binary data would be truncated in table 'tempdb.dbo.#B40707EB', column 'Email'. Truncated value: 'email1aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaadddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd'.
The statement has been terminated.